### ISSUE TITLE
[tests] PIP bits make Travis unreliable

### ISSUE THREAD

**Issue by bmr-cymru** [sosreport/sos#1270]
All the recent additions to the test infrastructure seem to have made Travis much less dependable, e..g:

```
Exception:
Traceback (most recent call last):
  File "/home/travis/virtualenv/python3.4.6/lib/python3.4/site-packages/pip/_vendor/requests/adapters.py", line 423, in send
    timeout=timeout
  File "/home/travis/virtualenv/python3.4.6/lib/python3.4/site-packages/pip/_vendor/requests/packages/urllib3/connectionpool.py", line 712, in urlopen
    release_conn=release_conn, **response_kw)
  File "/home/travis/virtualenv/python3.4.6/lib/python3.4/site-packages/pip/_vendor/requests/packages/urllib3/connectionpool.py", line 712, in urlopen
    release_conn=release_conn, **response_kw)
  File "/home/travis/virtualenv/python3.4.6/lib/python3.4/site-packages/pip/_vendor/requests/packages/urllib3/connectionpool.py", line 712, in urlopen
    release_conn=release_conn, **response_kw)
  File "/home/travis/virtualenv/python3.4.6/lib/python3.4/site-packages/pip/_vendor/requests/packages/urllib3/connectionpool.py", line 712, in urlopen
    release_conn=release_conn, **response_kw)
  File "/home/travis/virtualenv/python3.4.6/lib/python3.4/site-packages/pip/_vendor/requests/packages/urllib3/connectionpool.py", line 712, in urlopen
    release_conn=release_conn, **response_kw)
  File "/home/travis/virtualenv/python3.4.6/lib/python3.4/site-packages/pip/_vendor/requests/packages/urllib3/connectionpool.py", line 697, in urlopen
    retries = retries.increment(method, url, response=response, _pool=self)
  File "/home/travis/virtualenv/python3.4.6/lib/python3.4/site-packages/pip/_vendor/requests/packages/urllib3/util/retry.py", line 287, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
pip._vendor.requests.packages.urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='files.pythonhosted.org', port=443): Max retries exceeded with url: /packages/42/3f/669429ce58de2c22d8d2c542752e137ec4b9885fff398d3eceb1a7f5acb4/pep8-1.7.1-py2.py3-none-any.whl (Caused by ResponseError('too many 503 error responses',))
```

Is one of several, seemingly network condition related, exceptions from the last half hour. Having such a high false +ve/failure rate really starts to impact on the usefulness of this testing - I've had to restart almost every test manually today, and even then, they often fail on the retry.

**Comment by psachin** [comment]
@bmr-cymru No network issue(or failed test) since last seven days, I can only assume that it was due to travis <-> pythonhost.org network. Do you still want me to send PR? It will not be related to this issue, but it won't hurt to upgrade `pip` before installing requirements.

**Comment by BryanQuigley** [comment]
I just had a network failure in one of my tests today. :(

**Comment by BryanQuigley** [comment]
I've really only had to restart the python3.4 test - for whatever reason that's the one that fails the most (maybe least users?).

Ubuntu/Debian should no longer care about the python3.4 tests.  Does RH? 

For reference:
14.04 ESM uses Python 3.4, but we are not planning any new sosreport releases for it.  (Generally security/critical updates only)
16.04 LTS used Python 3.5 
18.04 LTS used Python 3.6
Next LTS and Debian stable: 3.7+

**Comment by BryanQuigley** [comment]
I believe this to be fixed with us dropping python 3.4 testing.  3.4 is not cached locally by travis so it was more likely to fail
